torch_xla/csrc/tensor_util.cpp

```diff
@@ -829,11 +882,15 @@ xla::PrimitiveType GetDevicePrimitiveType(xla::PrimitiveType type,
       if (UseBF16()) {
         return xla::PrimitiveType::BF16;
       }
+      if (UseF16()) {
+        return xla::PrimitiveType::F16;
+      }
       return xla_device.hw_type != DeviceType::TPU ? xla::PrimitiveType::F64
                                                    : xla::PrimitiveType::F32;
     case xla::PrimitiveType::F32:
-      // When PyTorch will support native BF16 type, the global configuration
-      // can be replaced (or augmented) with the proper mapping.
+      if (UseF16()) {
+        return xla::PrimitiveType::F16;
```

Comment:
Do we have a way to block user from setting both USE_F16 and USE_BF16 to 1? If user (mistakenly) do that, for F64 we will use BF16 and for F32 we will use F16. Might be better if we are consistently use one of them by switching the order of if check?

Reply:
That will result in errors in any case, as no platforms supports both 😉 
But for code consistency sake makes sense to flip the test.